Add new section with the scientific references (#1161)
TulipaEnergy/TulipaEnergyModel.jl: docs/src/30-concepts.md

!!! info "Are you into the math, objective function terms, and constraints?"
    The [`mathematical formulation`](@ref formulation) shows an overview of all the constraints and the variables in the model.

Another essential concept in the model is the [flexible time resolution](@ref flex-time-res), which allows for each asset to be considered in a single timestep (e.g., 1, 2, 3...) or in a range of timesteps (e.g., 1:3, meaning that the asset's variable represents the value of timesteps 1, 2, and 3). This concept allows the modeling of different dynamics depending on the asset; for instance, electricity assets can be modeled hourly, whereas hydrogen assets can be modeled in a 6-hour resolution (avoiding creating unnecessary constraints and variables).

The following sections explain these concepts in more detail.

> !!! info "Do you want to know more about the references used in this section?"

Reviewer: ```suggestion
!!! info "Want to know more about the references used in this section?"
```

>     Please visit the [scientific references](@ref scientific-refs) section to find links to the main papers that form the basis of this section.

Reviewer: ```suggestion
    Please visit the [scientific references](@ref scientific-refs) section for links to the papers that support this section.
```
